### DISCUSSION TITLE
Support more Launchpad Pro mk3 features

### DISCUSSION THREAD

**Discussion by on3iro** [helgoboss/helgobox#1109] [Ideas]
Hey :),

I just bought playtime 2 and am pretty excited about it. In the past I used to use a native instruments machine, but didn't like the way it integrated into DAWs. 
Currently I only have an Arturia Minilab MK2 and am thinking about replacing it with a launchpad X or pro.
However I wonder if it would be worth the cost, depending on the capabilities it brings in conjunction with playtime and realearn in Reaper.

I've already watched the playtime grid controller video, but there are still quite a few questions left unanswered:

1. Is it possible to duplicate clips to other slots, quantize them and also to change the fixed recording length from the launchpad pro?
2. Will I be able to use custom views on the launchpad e.g. to have a track volume section, a marco "encoder" section etc. (in Reaper as well as in playtime)
3. Will I still be able to use the launchpad as a keyboard/drumpad when it's connected with playtime? (it would be ideal if it would make use of the launchpad modes like session mode and note mode etc.)
4. Can I use the launchpad pros track selection buttons to select tracks and solo/mute/arm them?
5. Can the pro's step sequencer be used in any meaningful way
6. Depending on the capabilities: would you say the value of the Pro in conjunction with Playtime and Realearn is worth the cost over the X, or wouldn't I be able to use most of its features and would therefore be better off buying the X?

Thanks in advance for your support. Playtime is great and I really want to build a workflow around it :)

PS: If it would make sense we could also add this information directly to the playtime/realearn controller wiki sections, so it's easier to find in the future...

**Comment by helgoboss** [comment]
Hi!

> 1. Is it possible to duplicate clips to other slots, quantize them and also to change the fixed recording length from the launchpad pro?

Yes, that's possible and works out of the box.

However, if you would like to go with the Launchpad X, there are ways to make those things work there, too. You could modify the Launchpad X preset, remapping some buttons to execute different actions than indicated by the button label. Obviously, it wouldn't be as intuitive anymore, as the button label says something else, but it would work.

> 2. Will I be able to use custom views on the launchpad e.g. to have a track volume section, a marco "encoder" section etc. (in Reaper as well as in playtime)

No.

The following features are not yet supported: Note, Chord, Custom, Sequencer, Projects, Volume, Pan, Sends, Device, 
Swing, Tempo, Capture MIDI, Patterns, Steps, Pattern Settings, Velocity, Probability,
Mutation, Micro Step, Print to Clip.

This is also mentioned in the preset description (as shown in the video). Just now, I realized that it was not up-to-date and updated it.

I'm planning to add support for more features in the future, but I don't have an ETA right now. 

> 3. Will I still be able to use the launchpad as a keyboard/drumpad when it's connected with playtime? (it would be ideal if it would make use of the launchpad modes like session mode and note mode etc.)

Not yet. This is one of the features which are probably not difficult to add because it's essentially just switching a mode within the Launchpad ... but yeah, it's not there yet.

> 4. Can I use the launchpad pros track selection buttons to select tracks and solo/mute/arm them?

Yes, that's supported.

> 5. Can the pro's step sequencer be used in any meaningful way

I haven't tried. Probably also something that I would need to add by switching a mode.

> 6. Depending on the capabilities: would you say the value of the Pro in conjunction with Playtime and Realearn is worth the cost over the X, or wouldn't I be able to use most of its features and would therefore be better off buying the X?

Hmm, hard to say. It depends on what features you would use in practice and how much time you would be willing to wait for features that are not supported *yet*.

Some of those features which are not related to clip launching / session view are nowadays supported by the DrivenByMoss extension. But at the moment, it's an either-or thing. AFAIK, if you use DrivenByMoss, you can't use the controller for Playtime because DBM requires exclusive access to the device. In the future, I need to check whether it's easier to provide support for DrivenByMoss or to rebuild those things within Helgobox.

> PS: If it would make sense we could also add this information directly to the playtime/realearn controller wiki sections, so it's easier to find in the future...

Good idea.

**Reply by on3iro** [reply]
@helgoboss Thanks a lot for the quick response! Sounds like while It's already possible to do a few cool things with the launchpad it probably makes sense to wait a bit. Especially using it as an instrument and also for playtime would be really important, as I would like to only use a single controller and might even sell the minilab eventually. Also not being able to control VST encoders etc. with the pad would be a realy bummer (I currently use the minilabs encoders for this). Therefore I'll wait for now and keep an eye on updates ;)

**Reply by helgoboss** [reply]
True.

Personally, since I'm a piano player, I would always prefer encoders/knobs over pads for parameter control and keys over pads for playing instruments ... that's why it was not my priority so far. (And it's also the reason why I love the APC Key 25 mk2 much more than the Launchpad.)

But yes, I can see that those extra features are important for some people. I will try to look into it soon. Maybe there are a few quick wins in this area. I'm going to change the category of this discussion to "Ideas", which essentially turns it into a feature request.

**Reply by on3iro** [reply]
Awesome, I'll keep being subscribed to it then :D
